go-packages/meep-model/model.go

```diff
@@ -421,76 +421,203 @@ func (m *Model) UpdateNetChar(nc *dataModel.EventNetworkCharacteristicsUpdate) (
-	}
-	return false
-}
-
 // AddScenarioNode - Add scenario node
 func (m *Model) AddScenarioNode(node *dataModel.ScenarioNode) (err error) {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
 	if node == nil {
-		err = errors.New("node == nil")
-		return
+		return errors.New("node == nil")
 	}
 
-	// Find parent
+	// Find & validate parent
 	parentNode := m.nodeMap.FindByName(node.Parent)
 	if parentNode == nil {
-		err = errors.New("Parent element " + node.Parent + " not found in scenario " + m.name)
-		return
+		return errors.New("Parent element " + node.Parent + " not found in scenario " + m.name)
+	}
+	if !validateParentType(node.Type_, parentNode.nodeType) {
+		return errors.New("Invalid parent type: " + parentNode.nodeType + " for node type: " + node.Type_)
 	}
 
 	// Add element based on type
-	if node.Type_ == NodeTypeUE {
+	if isPhyLoc(node.Type_) {
 
 		// Get parent Network Location node & context information
-		if !m.isValidPoaNodeType(parentNode.nodeType) {
-			err = errors.New("Invalid parent type: " + parentNode.nodeType)
-			return
-		}
 		nl := parentNode.object.(*dataModel.NetworkLocation)
 
 		// Validate Physical Location
 		if node.NodeDataUnion == nil || node.NodeDataUnion.PhysicalLocation == nil {
-			err = errors.New("Missing Physical Location")
-			return
+			return errors.New("Missing Physical Location")
 		}
 		pl := node.NodeDataUnion.PhysicalLocation
-		err = validatePL(pl)
+		err = validatePhyLoc(pl)
 		if err != nil {
-			return
+			return err
 		}
 
 		// Make sure node Name is unique
 		n := m.nodeMap.FindByName(pl.Name)
 		if n != nil {
-			err = errors.New("Element " + pl.Name + " already exists in scenario " + m.name)
-			return
+			return errors.New("Element " + pl.Name + " already exists in scenario " + m.name)
 		}
 
-		// Remove any configured processes
+		// Ignore any configured processes
 		pl.Processes = make([]dataModel.Process, 0)
 
-		// Add PL to parent NL
+		// Add PhyLoc to parent NetLoc
 		nl.PhysicalLocations = append(nl.PhysicalLocations, *pl)
 
-		// Refresh node map
-		err = m.parseNodes()
+	} else if isProc(node.Type_) {
+
+		// Get parent Physical Location node & context information
+		pl := parentNode.object.(*dataModel.PhysicalLocation)
+
+		// Validate Process
+		if node.NodeDataUnion == nil || node.NodeDataUnion.Process == nil {
+			return errors.New("Missing Process")
+		}
+		proc := node.NodeDataUnion.Process
+		err = validateProc(proc)
 		if err != nil {
-			log.Error(err.Error())
+			return err
+		}
+
+		// Make sure node Name is unique
+		n := m.nodeMap.FindByName(proc.Name)
+		if n != nil {
+			return errors.New("Element " + proc.Name + " already exists in scenario " + m.name)
+		}
+
+		// Add Proc to parent PhyLoc
+		pl.Processes = append(pl.Processes, *proc)
+
+	} else {
+		return errors.New("Node type " + node.Type_ + " not supported")
+	}
+
+	// Refresh node map
+	err = m.parseNodes()
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	// Update scenario
+	err = m.refresh()
```

Comment:
if error check is not useful, just use _

Reply:
fixed code to return error